lib/hashie/mash.rb

```diff
@@ -346,7 +364,11 @@ def convert_value(val, duping = false) #:nodoc:
     private
 
     def log_built_in_message(method_key)
-      return if self.class.disable_warnings?
+      if self.class.disable_warnings_blacklist.any?
+        return if self.class.disable_warnings_blacklist.include?(method_key)
```

Comment:
The doubly-nested check can be eliminated because `include?(method_key)` for an empty blacklist will still be false, like the `any?` check.

Reply:
Oh, I remember why I did this now. We only check the blacklist _if_ there are blacklist items. Just `include?` will _always_ check it, and will ignore the second case, where we have an empty blacklist but `disable_warnings?` is true.

I'll see if changing the `disable_warnings?` signature helps with this.

Reply:
Both of these should be resolved by f967fa0c46dba0c8f4ccbffb5376438c7155af0c @michaelherold